client/domain/PrereqCheckFailureObjectFactory.js

```diff
@@ -197,10 +241,14 @@ tie.factory('PrereqCheckFailureObjectFactory', [
      *    unsupported libraries the user tried to import
      * @param {string} starterCode Should have the original code and function
      *    headers the user is given before they begin adding their own code.
+     * @param {string} wrongLangName Name of the error that corresponds to the
+     *    error information in WRONG_LANGUAGE_ERRORS
      * @returns {PrereqCheckFailure}
      */
-    PrereqCheckFailure.create = function(type, badImports, starterCode) {
-      return new PrereqCheckFailure(type, badImports, starterCode);
+    PrereqCheckFailure.create = function(
+        type, badImports, starterCode, wrongLangName) {
+      var name = (typeof wrongLangName === 'undefined') ? null : wrongLangName;
```

Comment:
I'm also seeing wrongLangErrorName, wrongLangName, and name. Let's be consistent and just use one of these throughout if they're all referring to the same concept. In particular "wrongLangName" conveys something quite different from "wrongLangErrorName" -- I thought it was referring to stuff like e.g. "c++".